Automattic/wp-calypso
# Add a dev helper for quick switching between OAuth and cookies auth

This PR adds a dev helper that show the current REST API authentication method (OAuth or cookies) and lets you switch with one click.

<img width="431" alt="Screenshot 2021-06-04 at 11 15 04" src="https://user-images.githubusercontent.com/664258/120778311-7c505280-c526-11eb-881a-2c48831f4fbc.png">

Under the hood, this simply changes the `oauth` config flag by reloading the page with a `?flags=[-]oauth` query param and storing the flag in session storage so that it persists across page reloads. I.e., it builds on previous work in #53006.

I'm also removing the special `/oauth-override` route from Jetpack Cloud and the `inJetpackCloudOAuthOverride` functionality. Now, the `oauth` config flag alone is the source of truth and we don't need any additional checks.

@cbauerman I believe that the new auth helper is a superior replacement for the `/oauth-override` endpoint which can be removed now. Please let me know if I'm mistaken about this 🙂 

**How to test:**
Open the `calypso.live` link for this PR in Safari or another cookie-blocking browser and verify that Calypso loads after switching to OAuth in the helper. (it should be showing only an empty placeholder with cookie auth)
